src/core/options-watcher.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+#pragma once
+
+
+#include <librealsense2/h/rs_option.h>
+#include <src/core/option-interface.h>
+
+#include <rsutils/signal.h>
+#include <rsutils/concurrency/concurrency.h>
+
+#include <map>
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <thread>
+
+
+namespace librealsense {
+
+
+// Automatically updates (queries) registered options values according to set time intervals
+// Can notify if a value have changed through a callback subscription.
+class options_watcher
+{
+public:
+    options_watcher( std::chrono::milliseconds update_interval = std::chrono::milliseconds( 1000 ) );
+    ~options_watcher();
+
+    // Register option to monitor value updates. Can set if update from HW (query) is necessary.
+    void register_option( rs2_option id, std::shared_ptr< option > option, bool update_from_hw = false );
+    void unregister_option( rs2_option id );
+
+    rsutils::subscription subscribe( std::function< void( const std::map< rs2_option, std::shared_ptr< option > > & ) > && callback );
```

Comment:
I suggest you do a (public) typedef for the callback function, so it's easy to reference here and elsewhere: `using callback = std::function< ... >;`

Reply:
Done